authentication/active_directory_authentication_provider.py

```diff
@@ -0,0 +1,46 @@
+#########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from ldap_authentication_provider import LDAPAuthenticationProvider
+
+DEFAULT_AUTH_ATTR_NAME = 'sAMAccountName'
+
+
+class ActiveDirectoryAuthenticationProvider(LDAPAuthenticationProvider):
+
+    def __init__(self, ldap_url, domain_name=None, search_properties=None):
+        super(ActiveDirectoryAuthenticationProvider, self)\
+            .__init__(ldap_url, search_properties)
+        self.domain_name = domain_name
+
+    def authenticate(self, userstore=None):
```

Comment:
Sadly, this can't be done because of the way the authentication method is being invoked